test/shared/utils.py

```diff
@@ -79,11 +80,21 @@ def expected_wheels(package_name, package_version, manylinux_versions=['manylinu
     # {distribution}-{version}(-{build tag})?-{python tag}-{abi tag}-{platform tag}.whl
     # {python tag} and {abi tag} are closely related to the python interpreter used to build the wheel
     # so we'll merge them below as python_abi_tag
-    python_abi_tags = ['cp27-cp27m', 'cp35-cp35m', 'cp36-cp36m', 'cp37-cp37m', 'cp38-cp38',
-                       'pp27-pypy_73', 'pp36-pypy36_pp73']
+
+    python_abi_tags = ['cp35-cp35m', 'cp36-cp36m', 'cp37-cp37m', 'cp38-cp38']
+    extra_python_abi_tags = ['cp27-cp27m', 'pp27-pypy_73', 'pp36-pypy36_pp73']
+
     if platform == 'linux':
-        python_abi_tags.append('cp27-cp27mu')  # python 2.7 has 2 different ABI on manylinux
-        architectures = {'cp': ['x86_64', 'i686'], 'pp': ['x86_64']}
+        if pm.machine() not in ['x86_64', 'i686']:
+            if manylinux_versions is None:
+                manylinux_versions = ['manylinux2014']
+            architectures = {'cp': [pm.machine()]}
+        else:
+            if manylinux_versions is None:
+                manylinux_versions = ['manylinux1', 'manylinux2010']
+            python_abi_tags += extra_python_abi_tags
+            python_abi_tags.append('cp27-cp27mu')  # python 2.7 has 2 different ABI on manylinux
+            architectures = {'cp': ['x86_64', 'i686'], 'pp': ['x86_64']}
```

Comment:
I might be wrong, but this feels like it could be somehow merged into a separate function with the large `if`-`elif`-`elif`-... from above, handling the translation of the platform to platform strings, and returning a list of the the platforms we are building? Maybe I'm wrong and it's not that straightforward, though. If I find the time, I'll give it a quick look myself.